### PR TITLE
package: libnss

### PR DESCRIPTION
**Pull request by mesaglio**
This is other 
This is other package needed to execute [cypress](https://docs.cypress.io/guides/getting-started/installing-cypress#UbuntuDebian) frontend testing tool.

**Important**: We need #7203 is merged to build this package.

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Comment by lpcalisi** [comment]
depends https://github.com/wolfi-dev/os/pull/7259

**Comment by mesaglio** [comment]
@mattmoor Here we found an issue in this compilation.

The `aarch64` compilation need a `CFLAG`, `-march`, with the value `armv8-a+crypto` to use that cpu feature. 
If we build an arm wolfi container de `build.sh` script work fine. 
We saw [melange force](https://github.com/wolfi-dev/os/blob/8174603e37ce4a322346e4e272731d38ea06e675/build-aarch64.env#L2) some params, and the final flags are `-march=armv8-a`.

Can you help with this?

ps: A workaround, only building en `x86` and `aarch64`, making an `if` in `nss.yaml` adding `CFLAGS=$CFALGS -march=armv8-a+crypto`. We dont believe this is the final solution for this problem.

cc @lpcalisi

File: nss.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      ./build.sh -c --disable-tests --opt --system-nspr --system-sqlite
+
+      mkdir -p "${{targets.destdir}}"/usr/lib
+      mkdir -p "${{targets.destdir}}"/usr/bin
+      mkdir -p "${{targets.destdir}}"/usr/include/nss
+
+      mv ../dist/Release/lib/*.so "${{targets.destdir}}"/usr/lib
+      mv ../dist/Release/bin/* "${{targets.destdir}}"/usr/bin
+      mv ../dist/public/nss/* "${{targets.destdir}}"/usr/include/nss
+
```
Comment (mattmoor) [inline nss.yaml:44]:
```suggestion

    - uses: strip

```

File: nss.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+var-transforms:
+  - from: ${{package.version}}
+    match: \.
+    replace: _
+    to: nss-package-version
+
+pipeline:
+  # we need to use runs because melange dont have a mercurial checkout pipeline
+  - runs: |
+      hg clone -r NSS_${{vars.nss-package-version}}_RTM https://hg.mozilla.org/projects/nss
```
Comment (mattmoor) [inline nss.yaml:33]: I'd split this into its own first step.

NB: As these are growing, we should consider adding a pipeline to checkout mercurial.  You can see examples of custom pipelines in `pipelines/`

**Comment by mattmoor** [comment]
I'll try to pull this down and experiment later, but have a few more meetings I need to get through.

File: nss.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+      mkdir -p "${{targets.destdir}}"/usr/lib
+      mkdir -p "${{targets.destdir}}"/usr/bin
+      mkdir -p "${{targets.destdir}}"/usr/include/nss
+
+      mv ../dist/Release/lib/*.so "${{targets.destdir}}"/usr/lib
+      mv ../dist/Release/bin/* "${{targets.destdir}}"/usr/bin
+      mv ../dist/public/nss/* "${{targets.destdir}}"/usr/include/nss
+
+    - uses: strip
```
Comment (mattmoor) [inline nss.yaml:45]: My suggestion was overly indented 🙃 
```suggestion
  - uses: strip
```

File: nss.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+var-transforms:
+  - from: ${{package.version}}
+    match: \.
+    replace: _
+    to: nss-package-version
+
+pipeline:
+  # we need to use runs because melange dont have a mercurial checkout pipeline
+  - runs: |
+      hg clone -r NSS_${{vars.nss-package-version}}_RTM https://hg.mozilla.org/projects/nss
```
Comment (mattmoor) [inline nss.yaml:33]: Actually, I realized that this is using a Mercurial ~ref, which isn't safe.  We need to reference deps by checksum.  I'm fixing `nspr` here: https://github.com/wolfi-dev/os/pull/7342

Comment (mattmoor) [inline nss.yaml:33]:
```suggestion
pipeline:
  - uses: fetch
    with:
      uri: https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_94_RTM/src/nss-${{package.version}}.tar.gz
      expected-sha256: 463ae180ee9e5ee9e3ad4f629326657e236780cc865572a930a16520abad9dd8

  - runs: |
```

Comment (mattmoor) [reply nss.yaml:33]: blah, maybe we still need the `var-transforms` for the `NSS_3_94_RTM` segment 🙃

**Comment by mattmoor** [comment]
h/t @kaniini https://github.com/wolfi-dev/os/pull/7358

File: nss.yaml (hunk 4 above)
Comment (mesaglio) [reply nss.yaml]: Change to use fetch

**Comment by mesaglio** [comment]
Hey @mattmoor appreciate you for the feedback!
I think this is ready to merge 🥳

**Comment by mattmoor** [comment]
I was sitting here wondering how the build env thing was picked up so quickly (thinking it was in `melange`), but it is in this repo! 🎉

**Review by mattmoor — APPROVED** [review]

**Pull request merged**